src/core/Directus/Application/Http/Middleware/ResponseCacheMiddleware.php

```diff
@@ -83,13 +83,11 @@ public function __invoke(Request $request, Response $response, callable $next)
                 case DirectusUserSessionsTableGateway::TOKEN_COOKIE :
                     $accessToken = decrypt_static_token($authorizationTokenObject['token']);
                     $userSession = $userSessionService->find(['token' => $accessToken]);
-                    if($userSession){
-                        $cookie = new Cookies();
-                        $cookie->set('session',['value' => $authorizationTokenObject['token'],'expires' =>$expiry->format(\DateTime::COOKIE),'path'=>'/','httponly' => true]);
-                        $response =  $response->withAddedHeader('Set-Cookie',$cookie->toHeaders());
-                    }else{
-                        $response =  $response->withoutHeader('Set-Cookie');
-                    }
+                    $cookie = new Cookies();
+                    $expiryAt = $userSession ? $expiry->format(\DateTime::COOKIE) : DateTimeUtils::now()->toString();
+                    $cookie->set('session',['value' => $authorizationTokenObject['token'],'expires' => $expiryAt ,'path'=>'/','httponly' => true]);
+                    $cookie->set('PHPSESSID',['expires' => DateTimeUtils::now()->toString()]);
```

Comment:
Do we need this @bjgajjar? Why do we have this? What does it do?

Reply:
`PHPSESSID` that is

Reply:
- We need this to update the `expired_at` time of `session` cookie in every `API` call.
- As you have recommended we don't need `PHPSESSID` as a cookie. So trying to remove it from the response.